FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,942 @@
+      fatalError("The default FirebaseApp instance must be configured before the " +
+        "default Remote Config instance can be initialized. One way to ensure " +
+        "this is to call `FirebaseApp.configure()` in the App Delegate's " +
+        "`application(_:didFinishLaunchingWithOptions:)` or the `@main` struct's " +
+        "initializer in SwiftUI.")
+    }
+
+    return remoteConfig(withFIRNamespace: firebaseNamespace, app: app)
+  }
+
+  // Use the provider to generate and return instances of FIRRemoteConfig for this specific app and
+  // namespace. This will ensure the app is configured before Remote Config can return an instance.
+  @objc(remoteConfigWithFIRNamespace:app:)
+  public static func remoteConfig(withFIRNamespace firebaseNamespace: String = RemoteConfigConstants
+    .NamespaceGoogleMobilePlatform,
+    app: FirebaseApp) -> RemoteConfig {
+    let provider = ComponentType<RemoteConfigInterop>
+      .instance(
+        for: RemoteConfigInterop.self,
+        in: app.container
+      ) as! any RemoteConfigProvider as RemoteConfigProvider
+    return provider.remoteConfig(forNamespace: firebaseNamespace)!
+  }
+
+  /// Last successful fetch completion time.
+  @objc public var lastFetchTime: Date? {
+    var fetchTime: Date?
+    queue.sync {
+      let lastFetchTimeInterval = self.settings.lastFetchTimeInterval
+      if lastFetchTimeInterval > 0 {
+        fetchTime = Date(timeIntervalSince1970: lastFetchTimeInterval)
+      }
+    }
+    return fetchTime
+  }
+
+  /// Last fetch status. The status can be any enumerated value from `RemoteConfigFetchStatus`.
+  @objc public var lastFetchStatus: RemoteConfigFetchStatus {
+    var currentStatus: RemoteConfigFetchStatus = .noFetchYet
+    queue.sync {
+      currentStatus = self.configFetch.settings.lastFetchStatus
+    }
+    return currentStatus
+  }
+
+  /// Config settings are custom settings.
+  @objc public var configSettings: RemoteConfigSettings {
+    get {
+      // These properties *must* be accessed and returned on the lock queue
+      // to ensure thread safety.
+      var minimumFetchInterval: TimeInterval = ConfigConstants.defaultMinimumFetchInterval
+      var fetchTimeout: TimeInterval = ConfigConstants.httpDefaultConnectionTimeout
+      queue.sync {
+        minimumFetchInterval = self.settings.minimumFetchInterval
+        fetchTimeout = self.settings.fetchTimeout
+      }
+
+      RCLog.debug("I-RCN000066",
+                  "Successfully read configSettings. Minimum Fetch Interval: " +
+                    "\(minimumFetchInterval), Fetch timeout: \(fetchTimeout)")
+      let settings = RemoteConfigSettings()
+      settings.minimumFetchInterval = minimumFetchInterval
+      settings.fetchTimeout = fetchTimeout
+      RCLog.debug("I-RCN987366",
+                  "Successfully read configSettings. Minimum Fetch Interval: " +
+                    "\(minimumFetchInterval), Fetch timeout: \(fetchTimeout)")
+      return settings
```

Comment:
The ObjC implementation would recreate a network session before returning the settings.

```objc
  /// The NSURLSession needs to be recreated whenever the fetch timeout may be updated.
  [_configFetch recreateNetworkSession];
  return settings;
```

Reply:
I don't understand why the `get` needs to do this, but will add it back for now to be consistent.